memory/allocator.go

```diff
@@ -19,6 +19,14 @@ var _ memory.Allocator = (*Allocator)(nil)
 
 // Allocator tracks the amount of memory being consumed by a query.
 type Allocator struct {
+	// Variables accessed with atomic operations should be at
+	// the beginning of the struct to ensure byte alignment is correct.
```

Comment:
I think a link to this https://golang.org/pkg/sync/atomic/#pkg-note-BUG in the comment would be helpful.

Reply:
Good idea. I didn't even know this note existed but it's perfect for the comment.